winterfell/app.js

```diff
@@ -1,47 +1,60 @@
 #!/usr/bin/env node
+var Biscuit = require('./services/biscuit');
+var bunyan = require('bunyan');
+var Constants = require('./utils/constants');
+var documentRenderingConfig = require('./config/documentRendering');
 var express = require('express');
-var session = require('express-session');
 var fs = require('fs');
+var Log = require('./utils/logHelper');
 var path = require('path');
-var bodyParser = require('body-parser');
-var Constants = require('./utils/constants');
-var ENVIRONMENT = Constants.environment;
-var documentRenderingConfig = require('./config/documentRendering');
+var session = require('express-session');
 
+var ENVIRONMENT = Constants.environment;
 var environment = process.env.NODE_ENV || ENVIRONMENT.LOCAL;
-var Biscuit = require('./services/biscuit');
 
+// Initialize App
 var app = express();
-app.set('secretsFile', require('./config/biscuit')[environment]);
-var biscuit = new Biscuit(app);
-
 app.environment = environment;
 app.baseUrl = Constants.baseUrl[app.environment];
 
-// Setup biscuit keys
-function setupBiscuitKey(keyName) {
-  var yamlKey = Constants.biscuitKeys[keyName];
-  var secret = biscuit.get(environment + '::' + yamlKey);
-  app.set(keyName, secret);
-}
-
-setupBiscuitKey(Constants.KEY_LOB_API);
-setupBiscuitKey(Constants.KEY_IDME_CLIENT_ID);
-setupBiscuitKey(Constants.KEY_IDME_SECRET_ID);
+// Initialize Logger
+var log = bunyan.createLogger({
+  name: 'vetafi_' + environment,
+  streams: [{
+    path: '../logs/app.js' // todo: these logs need to be moved into the same level
+  }]
+});
+app.log = log;                                            // Main logger
+app.logApi = app.log.child({widget_type: 'api'});         // Logger for API request/responses
+app.log.info("New logger created.");
 
+// Initialize Node Modules`
 function loadIntoBuild (app, targetDir) {
   var normalizedPath = path.join(__dirname, targetDir);
   fs.readdirSync(normalizedPath).forEach(function (file) {
     require(normalizedPath + '/' + file)(app);
   });
   return app;
 }
-
 loadIntoBuild(app, 'utils');
 loadIntoBuild(app, 'middlewares');
 loadIntoBuild(app, 'services');
 loadIntoBuild(app, 'controllers');
+Log.console("Node modules loaded.");
+
+// Initialize Biscuit
+app.set('secretsFile', require('./config/biscuit')[environment]);
```

Comment:
just a reordering of operations. Wanted to log what initializing steps we passed in case it failed somewhere in the middle. That would mean logging would have to be first.
For biscuit, does the order matter?

On second thought it does since the app keys depend on biscuit to be set first. My bad, will swap the ordering so that biscuit comes before loading modules.